### PR TITLE
Density, viscosity and pressure as output variables

### PR DESCRIPTION
**Pull request by mschreter**

### CODE REVIEW

**Comment by mschreter** [comment]
The values on ghost dofs are doubled at the moment:

density

**Comment by peterrum** [comment]
I am in a telco. I'll write you in 5 minutes.

File: include/meltpooldg/flow/two_phase_flow_problem.hpp [hunk 1]
```diff
@@ -381,6 +361,68 @@ namespace MeltPoolDG
...
+
+        AlignedVector<VectorizedArray<double>> prolongation_matrix_1d(
+          fe_fine.dofs_per_cell * fe_coarse.dofs_per_cell);
+
+        for (unsigned int i = 0, k = 0; i < fe_coarse.dofs_per_cell; ++i)
+          for (unsigned int j = 0; j < fe_fine.dofs_per_cell; ++j, ++k)
+            prolongation_matrix_1d[k] = matrix(j, i);
+
+        scratch_data->get_matrix_free().template cell_loop<VectorType, std::nullptr_t>(
+          [&](const auto &matrix_free, auto &vec, const auto &, auto macro_cells) {
```
Comment (peterrum) [inline include/meltpooldg/flow/two_phase_flow_problem.hpp:383]: Oh the reason is that `MatrixFree` performs `compress()` on the destination vector at the end. I am afraid we cannot turn it of in the case of a `MatrixFree::cell_loop()`. What we have to do is to loop over all macro cells on our own

```cpp
for(unsigned int cell = 0; cell < matrix_free.n_macro_cells(); ++cell)
{
  // from here the code should be identical!
}
```

**Comment by mschreter** [comment]
@peterrum I would be ready too merge if you agree :).

**Review by peterrum — APPROVED** [review]
I took the liberty to fix the typo. I'll merge once CI is happy!

**Pull request merged**